Source/Core/Core/HW/DSP.cpp

```diff
@@ -474,6 +471,8 @@ void UpdateAudioDMA()
     // Read audio at g_audioDMA.current_source_address in RAM and push onto an
     // external audio fifo in the emulator, to be mixed with the disc
     // streaming output.
+    void* address = Memory::GetPointer(s_audioDMA.current_source_address);
+    AudioCommon::SendAIBuffer((short*)address, 8);
```

Comment:
If you're already moving this, make this a `reinterpret_cast`. It's still UB but at least it's more obvious that way.

If it doesn't cause any issues (not to familiar with this code, unclear if the pointer needs to stay valid after the call is done) you might also just get rid of the UB by making a stack-allocated `std::array<short, 8>` and memcpy'ing into that before passing to `SendAIBuffer()`.

Reply:
The pointer only remains valid while the call is in progress; `SendAIBuffer` copies to its own buffer.

I'm not sure if an extra copy is a good idea if it's not strictly necessary, since this is called 4000 times a second.  It might be better to pass `void*` to `SendAIBuffer` and have it deal with converting from an audio block to samples (which does include swapping from big endian to little endian), but the actual code is a bit complicated since the endian conversion is performed on a separate thread in `MixerFifo::Mix` and doesn't apply to everything (the Wiimote and GBA speakers are already little endian).

I'll do `reinterpret_cast` for now.